--- a/Source/exehead/fileform.h
+++ b/Source/exehead/fileform.h
@@ -51,155 +51,137 @@
 {
   EW_INVALID_OPCODE,    // zero is invalid. useful for catching errors. (otherwise an all zeroes instruction
                         // does nothing, which is easily ignored but means something is wrong)
-  EW_RET,               // return from function call
-  EW_NOP,               // Nop/Jump, do nothing: 1, [?new address+1:advance one]
-  EW_ABORT,             // Abort: 1 [status]
+#ifdef NSIS_SUPPORT_ACTIVEXREG
+  EW_REGISTERDLL,       // Register DLL: 3,[DLL file name, string ptr of function to call, text to put in display (<0 if none/pass parms), 1 - no unload, 0 - unload]
+#endif
+#ifdef NSIS_SUPPORT_CREATESHORTCUT
+  EW_CREATESHORTCUT,    // Make Shortcut: 5, [link file, target file, parameters, icon file, packed CS_*]
+#endif
+#ifdef NSIS_LOCKWINDOW_SUPPORT
+  EW_LOCKWINDOW,
+#endif
+#ifdef NSIS_SUPPORT_REGISTRYFUNCTIONS
+  EW_DELREG,            // DeleteRegValue/DeleteRegKey: 4, [root key(int), KeyName, ValueName, ActionAndFlags(DELREG*)]
+  EW_WRITEREG,          // Write Registry value: 5, [RootKey(int),KeyName,ItemName,ItemData,typelen]
+                        //  typelen=1 for str, 2 for dword, 3 for binary, 0 for expanded str
+  EW_READREGSTR,        // ReadRegStr: 5 [output, rootkey(int), keyname, itemname, ==1?int::str]
+  EW_REGENUM,           // RegEnum: 5 [output, rootkey, keyname, index, ?key:value]
+#endif
   EW_QUIT,              // Quit: 0
-  EW_CALL,              // Call: 1 [new address+1]
-  EW_UPDATETEXT,        // Update status text: 2 [update str, ui_st_updateflag=?ui_st_updateflag:this]
-  EW_SLEEP,             // Sleep: 1 [sleep time in milliseconds]
+#ifdef NSIS_CONFIG_LOG
+  EW_LOG,               // LogText: 2 [0, text] / LogSet: [1, logstate]
+#endif
   EW_BRINGTOFRONT,      // BringToFront: 0
-  EW_CHDETAILSVIEW,     // SetDetailsView: 2 [listaction,buttonaction]
   EW_SETFILEATTRIBUTES, // SetFileAttributes: 2 [filename, attributes]
-  EW_CREATEDIR,         // Create directory: 2, [path, ?update$INSTDIR, ?restrictAcl]
+#ifdef NSIS_SUPPORT_COPYFILES
+  EW_COPYFILES,         // CopyFiles: 3 [source mask, destination location, flags]
+#endif
+  EW_GETFLAG,           // Gets a flag: 2 [output, id]
+#ifdef NSIS_SUPPORT_EXECUTE
+  EW_EXECUTE,           // Execute program: 3,[complete command line,waitflag,>=0?output errorcode]
+#endif
   EW_IFFILEEXISTS,      // IfFileExists: 3, [file name, jump amount if exists, jump amount if not exists]
-  EW_SETFLAG,           // Sets a flag: 2 [id, data]
+#ifdef NSIS_SUPPORT_RMDIR
+  EW_RMDIR,             // RMDir: 2 [path, recursiveflag]
+#endif
+  EW_SLEEP,             // Sleep: 1 [sleep time in milliseconds]
+  EW_ABORT,             // Abort: 1 [status]
+  EW_RET,               // return from function call
+#ifdef NSIS_SUPPORT_REBOOT
+  EW_REBOOT,            // Reboot: 0
+#endif
+#ifdef NSIS_SUPPORT_ENVIRONMENT
+  EW_READENVSTR,        // ReadEnvStr/ExpandEnvStrings: 3 [output, string_with_env_variables, IsRead]
+#endif
+#ifdef NSIS_SUPPORT_MESSAGEBOX
+  EW_MESSAGEBOX,        // MessageBox: 5,[MB_flags,text,retv1:retv2,moveonretv1:moveonretv2]
+#endif
+#ifdef NSIS_SUPPORT_DELETE
+  EW_DELETEFILE,        // Delete File: 2, [filename, rebootok]
+#endif
+#ifdef NSIS_SUPPORT_INIFILES
+  EW_READINISTR,        // ReadINIStr: 4 [output, section, name, ini_file]
+  EW_WRITEINI,          // Write INI String: 4, [Section, Name, Value, INI File]
+#endif
   EW_IFFLAG,            // If a flag: 4 [on, off, id, new value mask]
-  EW_GETFLAG,           // Gets a flag: 2 [output, id]
 #ifdef NSIS_SUPPORT_RENAME
   EW_RENAME,            // Rename: 3 [old, new, rebootok]
 #endif
+  EW_CALL,              // Call: 1 [new address+1]
+  EW_SETFLAG,           // Sets a flag: 2 [id, data]
+  EW_CREATEDIR,         // Create directory: 2, [path, ?update$INSTDIR, ?restrictAcl]
 #ifdef NSIS_SUPPORT_FNUTIL
-  EW_GETFULLPATHNAME,   // GetFullPathName: 2 [output, input, ?lfn:sfn]
   EW_SEARCHPATH,        // SearchPath: 2 [output, filename]
+  EW_GETFULLPATHNAME,   // GetFullPathName: 2 [output, input, ?lfn:sfn]
   EW_GETTEMPFILENAME,   // GetTempFileName: 2 [output, base_dir]
 #endif
+  EW_UPDATETEXT,        // Update status text: 2 [update str, ui_st_updateflag=?ui_st_updateflag:this]
+#ifdef NSIS_SUPPORT_FILEFUNCTIONS
+  EW_FSEEK,             // FileSeek: 4  [handle, offset, mode, >=0?positionoutput]
+  EW_FCLOSE,            // FileClose: 1 [handle]
+  EW_FOPEN,             // FileOpen: 4  [name, openmode, createmode, outputhandle]
+  EW_FPUTS,             // FileWrite: 3 [handle, string, ?int:string]
+  EW_FGETS,             // FileRead: 4  [handle, output, maxlen, ?getchar:gets]
+#endif//NSIS_SUPPORT_FILEFUNCTIONS
+  EW_GETOSINFO,         // 1+ [operation, ...]
+  EW_CHDETAILSVIEW,     // SetDetailsView: 2 [listaction,buttonaction]
 #ifdef NSIS_SUPPORT_FILE
   EW_EXTRACTFILE,       // File to extract: 6 [overwriteflag, output filename, compressed filedata, filedatetimelow, filedatetimehigh, allow ignore]
                         //  overwriteflag: 0x1 = no. 0x0=force, 0x2=try, 0x3=if date is newer
 #endif
-#ifdef NSIS_SUPPORT_DELETE
-  EW_DELETEFILE,        // Delete File: 2, [filename, rebootok]
+#ifdef NSIS_CONFIG_COMPONENTPAGE
+  EW_INSTTYPESET,       // InstTypeSetFlags:  3: [idx, 0, flags]
+                        // InstTypeGetFlags:  3: [idx, 1, output]
+  EW_SECTIONSET,        // SectionSetText:    3: [idx, 0, text]
+                        // SectionGetText:    3: [idx, 1, output]
+                        // SectionSetFlags:   3: [idx, 2, flags]
+                        // SectionGetFlags:   3: [idx, 3, output]
 #endif
-#ifdef NSIS_SUPPORT_MESSAGEBOX
-  EW_MESSAGEBOX,        // MessageBox: 5,[MB_flags,text,retv1:retv2,moveonretv1:moveonretv2]
+#ifdef NSIS_CONFIG_ENHANCEDUI_SUPPORT
+  EW_SHOWWINDOW,        // ShowWindow:        2: [hwnd, show state]
+  EW_LOADANDSETIMAGE,   // SetBrandingImage/LoadAndSetImage: 5: [ctrl imagetype lrflags imageid [output]]
+  EW_CREATEFONT,        // CreateFont:        5: [handle output, face name, height, weight, flags]
+  EW_SETCTLCOLORS,      // SerCtlColors:      3: [hwnd, pointer to struct colors]
+  EW_GETDLGITEM,        // GetDlgItem:        3: [outputvar, dialog, item_id]
 #endif
-#ifdef NSIS_SUPPORT_RMDIR
-  EW_RMDIR,             // RMDir: 2 [path, recursiveflag]
-#endif
-#ifdef NSIS_SUPPORT_STROPTS
-  EW_STRLEN,            // StrLen: 2 [output, input]
-  EW_ASSIGNVAR,         // Assign: 4 [variable (0-9) to assign, string to assign, maxlen, startpos]
-  EW_STRCMP,            // StrCmp: 5 [str1, str2, jump_if_equal, jump_if_not_equal, case-sensitive?]
-#endif
-#ifdef NSIS_SUPPORT_ENVIRONMENT
-  EW_READENVSTR,        // ReadEnvStr/ExpandEnvStrings: 3 [output, string_with_env_variables, IsRead]
-#endif
 #ifdef NSIS_SUPPORT_INTOPTS
   EW_INTCMP,            // IntCmp: 6 [val1, val2, equal, val1<val2, val1>val2, flags] where flags: bit 0x01 is set for unsigned operations and bit 0x8000 is set for 64-bit operations
-  EW_INTOP,             // IntOp: 4 [output, input1, input2, op] where op: 0=add, 1=sub, 2=mul, 3=div, 4=bor, 5=band, 6=bxor, 7=bnot input1, 8=lor, 9=land 10=mod, 11=shl, 12=sar, 13=shr (bneg is implemented with bxor in compiler)
   EW_INTFMT,            // IntFmt: 4 [output, format, input, 64-bit]
+  EW_INTOP,             // IntOp: 4 [output, input1, input2, op] where op: 0=add, 1=sub, 2=mul, 3=div, 4=bor, 5=band, 6=bxor, 7=bnot input1, 8=lor, 9=land 10=mod, 11=shl, 12=sar, 13=shr (bneg is implemented with bxor in compiler)
 #endif
-#ifdef NSIS_SUPPORT_STACK
-  EW_PUSHPOP,           // Push/Pop/Exchange: 3 [variable/string, ?pop:push, ?exch]
-#endif
 #ifdef NSIS_SUPPORT_HWNDS
-  EW_FINDWINDOW,        // FindWindow: 5, [outputvar, window class,window name, window_parent, window_after]
   EW_SENDMESSAGE,       // SendMessage: 6 [output, hwnd, msg, wparam, lparam, [wparamstring?1:0 | lparamstring?2:0 | timeout<<2]
   EW_ISWINDOW,          // IsWindow: 3 [hwnd, jump_if_window, jump_if_notwindow]
+  EW_FINDWINDOW,        // FindWindow: 5, [outputvar, window class,window name, window_parent, window_after]
 #endif
-
-#ifdef NSIS_CONFIG_ENHANCEDUI_SUPPORT
-  EW_GETDLGITEM,        // GetDlgItem:        3: [outputvar, dialog, item_id]
-  EW_SETCTLCOLORS,      // SerCtlColors:      3: [hwnd, pointer to struct colors]
-  EW_LOADANDSETIMAGE,   // SetBrandingImage/LoadAndSetImage: 5: [ctrl imagetype lrflags imageid [output]]
-  EW_CREATEFONT,        // CreateFont:        5: [handle output, face name, height, weight, flags]
-  EW_SHOWWINDOW,        // ShowWindow:        2: [hwnd, show state]
+#ifdef NSIS_SUPPORT_STROPTS
+  EW_STRLEN,            // StrLen: 2 [output, input]
+  EW_ASSIGNVAR,         // Assign: 4 [variable (0-9) to assign, string to assign, maxlen, startpos]
+  EW_STRCMP,            // StrCmp: 5 [str1, str2, jump_if_equal, jump_if_not_equal, case-sensitive?]
 #endif
-
-#ifdef NSIS_SUPPORT_SHELLEXECUTE
-  EW_SHELLEXEC,         // ShellExecute program: 5, [SEE_MASK_FLAG_*, verb, file, parameters, showwindow] (Will wait if SEE_MASK_NOCLOSEPROCESS is set)
-#endif
-
-#ifdef NSIS_SUPPORT_EXECUTE
-  EW_EXECUTE,           // Execute program: 3,[complete command line,waitflag,>=0?output errorcode]
-#endif
-
 #ifdef NSIS_SUPPORT_GETFILETIME
   EW_GETFILETIME,       // GetFileTime; 3 [file highout lowout]
 #endif
-
-#ifdef NSIS_SUPPORT_GETDLLVERSION
-  EW_GETDLLVERSION,     // GetDLLVersion: 4 [file highout lowout fixedoffset]
+#ifdef NSIS_SUPPORT_STACK
+  EW_PUSHPOP,           // Push/Pop/Exchange: 3 [variable/string, ?pop:push, ?exch]
 #endif
-
-#ifdef NSIS_SUPPORT_ACTIVEXREG
-  EW_REGISTERDLL,       // Register DLL: 3,[DLL file name, string ptr of function to call, text to put in display (<0 if none/pass parms), 1 - no unload, 0 - unload]
+#ifdef NSIS_CONFIG_UNINSTALL_SUPPORT
+  EW_WRITEUNINSTALLER,  // WriteUninstaller: 3 [name, offset, icon_size]
 #endif
-
-#ifdef NSIS_SUPPORT_CREATESHORTCUT
-  EW_CREATESHORTCUT,    // Make Shortcut: 5, [link file, target file, parameters, icon file, packed CS_*]
-#endif
-
-#ifdef NSIS_SUPPORT_COPYFILES
-  EW_COPYFILES,         // CopyFiles: 3 [source mask, destination location, flags]
-#endif
-
-#ifdef NSIS_SUPPORT_REBOOT
-  EW_REBOOT,            // Reboot: 0
-#endif
-
-#ifdef NSIS_SUPPORT_INIFILES
-  EW_WRITEINI,          // Write INI String: 4, [Section, Name, Value, INI File]
-  EW_READINISTR,        // ReadINIStr: 4 [output, section, name, ini_file]
-#endif
-
-#ifdef NSIS_SUPPORT_REGISTRYFUNCTIONS
-  EW_DELREG,            // DeleteRegValue/DeleteRegKey: 4, [root key(int), KeyName, ValueName, ActionAndFlags(DELREG*)]
-  EW_WRITEREG,          // Write Registry value: 5, [RootKey(int),KeyName,ItemName,ItemData,typelen]
-                        //  typelen=1 for str, 2 for dword, 3 for binary, 0 for expanded str
-  EW_READREGSTR,        // ReadRegStr: 5 [output, rootkey(int), keyname, itemname, ==1?int::str]
-  EW_REGENUM,           // RegEnum: 5 [output, rootkey, keyname, index, ?key:value]
-#endif
-
-#ifdef NSIS_SUPPORT_FILEFUNCTIONS
-  EW_FCLOSE,            // FileClose: 1 [handle]
-  EW_FOPEN,             // FileOpen: 4  [name, openmode, createmode, outputhandle]
-  EW_FPUTS,             // FileWrite: 3 [handle, string, ?int:string]
-  EW_FGETS,             // FileRead: 4  [handle, output, maxlen, ?getchar:gets]
-  EW_FSEEK,             // FileSeek: 4  [handle, offset, mode, >=0?positionoutput]
-#endif//NSIS_SUPPORT_FILEFUNCTIONS
-
+  EW_RESERVEDOPCODE,    // Free slot, feel free to use it for something
 #ifdef NSIS_SUPPORT_FINDFIRST
-  EW_FINDCLOSE,         // FindClose: 1 [handle]
   EW_FINDNEXT,          // FindNext: 2  [output, handle]
   EW_FINDFIRST,         // FindFirst: 2 [filespec, output, handleoutput]
+  EW_FINDCLOSE,         // FindClose: 1 [handle]
 #endif
-
-#ifdef NSIS_CONFIG_UNINSTALL_SUPPORT
-  EW_WRITEUNINSTALLER,  // WriteUninstaller: 3 [name, offset, icon_size]
+#ifdef NSIS_SUPPORT_GETDLLVERSION
+  EW_GETDLLVERSION,     // GetDLLVersion: 4 [file highout lowout fixedoffset]
 #endif
-
-#ifdef NSIS_CONFIG_LOG
-  EW_LOG,               // LogText: 2 [0, text] / LogSet: [1, logstate]
+#ifdef NSIS_SUPPORT_SHELLEXECUTE
+  EW_SHELLEXEC,         // ShellExecute program: 5, [SEE_MASK_FLAG_*, verb, file, parameters, showwindow] (Will wait if SEE_MASK_NOCLOSEPROCESS is set)
 #endif
+  EW_NOP,               // Nop/Jump, do nothing: 1, [?new address+1:advance one]
 
-#ifdef NSIS_CONFIG_COMPONENTPAGE
-  EW_SECTIONSET,        // SectionSetText:    3: [idx, 0, text]
-                        // SectionGetText:    3: [idx, 1, output]
-                        // SectionSetFlags:   3: [idx, 2, flags]
-                        // SectionGetFlags:   3: [idx, 3, output]
-  EW_INSTTYPESET,       // InstTypeSetFlags:  3: [idx, 0, flags]
-                        // InstTypeGetFlags:  3: [idx, 1, output]
-#endif
-
-  EW_GETOSINFO,         // 1+ [operation, ...]
-  EW_RESERVEDOPCODE,    // Free slot, feel free to use it for something
-
-#ifdef NSIS_LOCKWINDOW_SUPPORT
-  EW_LOCKWINDOW,
-#endif
-
 #ifdef _UNICODE     // opcodes available only in Unicode installers must be at the end of the enumeration
 #ifdef NSIS_SUPPORT_FILEFUNCTIONS
   EW_FPUTWS,            // FileWriteUTF16LE: 4 [handle, string, ?int:string, TryWriteBOM]
@@ -227,7 +209,7 @@
-#define FH_SIG 0xDEADBEEF
+#define FH_SIG 0x33323337
 
 // neato surprise signature that goes in firstheader. :)
-#define FH_INT1 0x6C6C754E
+#define FH_INT1 0x6372614D
 #define FH_INT2 0x74666F73
 #define FH_INT3 0x74736E49
 
